.drone.yml

```diff
@@ -1,124 +1,151 @@
-    commands:
-      - docker login -u="ukhomeofficedigital+drone" -p=$${DOCKER_PASSWORD} quay.io
-      - docker tag app-$${DRONE_COMMIT_SHA} quay.io/ukhomeofficedigital/brpapp:${DRONE_COMMIT_SHA}
-      - docker push quay.io/ukhomeofficedigital/brpapp:$${DRONE_COMMIT_SHA}
-    when:
-      branch: master
-      event: push
-
-  trigger_deploy_to_brp_dev:
-    image: quay.io/ukhomeofficedigital/drone-trigger:v0.3.0
-    secrets:
-      - DRONE_TOKEN
-    drone_server: https://drone.acp.homeoffice.gov.uk
-    repo: UKHomeOffice/kube-brpapp
+- name: clone_repo
+  pull: if-not-exists
+  image: docker:git
+  environment:
+    DRONE_GIT_USERNAME:
+      from_secret: DRONE_GIT_USERNAME
+    DRONE_GIT_TOKEN:
+      from_secret: DRONE_GIT_TOKEN
+  commands:
+    - git clone https://$${DRONE_GIT_USERNAME}:$${DRONE_GIT_TOKEN}@github.com/UKHomeOffice/kube-brpapp.git
+  when:
     branch: master
-    deploy_to: dev
-    params: BRPAPP_IMAGE_TAG=${DRONE_COMMIT_SHA},KUBE_NAMESPACE=brp-dev
-    when:
-      branch: master
-      event: push
+    event: push
 
-  trigger_deploy_to_brp_preprod:
-    image: quay.io/ukhomeofficedigital/drone-trigger:v0.3.0
-    secrets:
-      - DRONE_TOKEN
-    drone_server: https://drone.acp.homeoffice.gov.uk
-    repo: UKHomeOffice/kube-brpapp
+- name: deploy_to_dev
+  pull: if-not-exists
+  image: quay.io/ukhomeofficedigital/kd:v1.14.0
+  commands:
+    - cd kube-brpapp
+    - bin/deploy.sh
+  environment:
+    KUBE_SERVER_DEV:
+      from_secret: kube_server_dev
+    KUBE_TOKEN_DEV:
+      from_secret: kube_token_dev
+    KUBE_NAMESPACE: brp-dev
+    APP_IMAGE_TAG: ${DRONE_COMMIT_SHA}
+  when:
     branch: master
-    deploy_to: preprod
-    params: BRPAPP_IMAGE_TAG=${DRONE_COMMIT_SHA},KUBE_NAMESPACE=brp-preprod
-    when:
-      event: deployment
-      environment: preprod
+    event: push
 
-  trigger_deploy_to_brp_prod:
-    image: quay.io/ukhomeofficedigital/drone-trigger:v0.3.0
-    secrets:
-      - DRONE_TOKEN
-    drone_server: https://drone.acp.homeoffice.gov.uk
-    repo: UKHomeOffice/kube-brpapp
-    branch: master
-    deploy_to: prod
-    params: BRPAPP_IMAGE_TAG=${DRONE_COMMIT_SHA},KUBE_NAMESPACE=brp
-    when:
-      event: deployment
-      environment: prod
+- name: deploy_to_preprod
+  pull: if-not-exists
+  image: quay.io/ukhomeofficedigital/kd:v1.14.0
+  commands:
+    - cd kube-brpapp
+    - bin/deploy.sh
+  environment:
+    KUBE_SERVER_DEV:
+      from_secret: kube_server_dev
+    KUBE_TOKEN_DEV:
+      from_secret: kube_token_dev
+    KUBE_NAMESPACE: brp-preprod
+    APP_IMAGE_TAG: ${DRONE_COMMIT_SHA}
+  when:
+    event: promote
+    target: preprod
+
+- name: deploy_to_prod
+  pull: if-not-exists
+  image: quay.io/ukhomeofficedigital/kd:v1.14.0
+  commands:
+    - cd kube-brpapp
+    - bin/deploy.sh
+  environment:
+    KUBE_SERVER_PROD:
+      from_secret: kube_server_prod
+    KUBE_TOKEN_PROD:
+      from_secret: kube_token_prod
+    KUBE_NAMESPACE: brp
+    APP_IMAGE_TAG: ${DRONE_COMMIT_SHA}
```

Comment:
Whats this used for?  Althoug maybe its in the deployment files in whihc case leave it alone

Reply:
KUBE_NAMESPACE is used the the deployment.yaml file, and APP_IMAGE_TAG has been changed to "BRPAPP_IMAGE_TAG", and is present in app-deployment.yaml.